src/Pmi/Application/AbstractApplication.php

```diff
@@ -294,29 +306,21 @@ public function getGoogleLogoutUrl($route = 'home')
     {
         $dest = $this->generateUrl($route, [], true);
 
-        if ($this->getConfig('gae_auth')) {
-            $cls = $this->getGoogleServiceClass();
-            return class_exists($cls) ? $cls::createLogoutURL($dest) : null;
+        if ($this->isLocal() && $this->getConfig('gae_auth')) {
+            return $this['isUnitTest'] ? null : $dest;
         } else {
             // http://stackoverflow.com/a/14831349/1402028
             return "https://www.google.com/accounts/Logout?continue=https://appengine.google.com/_ah/logout?continue=$dest";
         }
     }
 
-    public function getGoogleLoginUrl($route = 'home')
-    {
-        $dest = $this->generateUrl($route, [], true);
-
-        if ($this->getConfig('gae_auth')) {
-            $cls = $this->getGoogleServiceClass();
-            return class_exists($cls) ? $cls::createLoginURL($dest) : null;
-        }
-    }
-
     public function getUser()
     {
         $token = $this['security.token_storage']->getToken();
-        return $token ? $token->getUser() : null;
+        if ($token) {
+            return is_object($token->getUser()) ? $token->getUser() : null;
+        }
+        return null;
```

Comment:
I think this is a bit easier to read:

```suggestion
        if ($token && is_object($token->getUser())) {
            return $token->getUser();
        } else {
            return null;
        }
```